eZ/Publish/Core/Persistence/Cache/UrlAliasHandler.php

```diff
@@ -79,7 +79,15 @@ public function createCustomUrlAlias($locationId, $path, $forwarding = false, $l
             $alwaysAvailable
         );
 
-        $this->cache->invalidateTags(['urlAlias-location-' . $locationId, 'urlAlias-location-path-' . $locationId, 'urlAlias-notFound']);
+        $this->cache->invalidateTags([
+            'urlAlias-location-' . $locationId,
+            'urlAlias-location-path-' . $locationId,
+            'urlAlias-notFound',
+        ]);
+
+        if ($urlAlias instanceof UrlAlias) {
+            $this->cache->invalidateTags(['urlAlias-' . $urlAlias->id]);
+        }
```

Comment:
Like any other calls to cache, they involve a roundtrip to the server, so good practice is to build up the keys or tags to clear before calling it. If so it would be:

```suggestion
        $cacheTags = [
            'urlAlias-location-' . $locationId,
            'urlAlias-location-path-' . $locationId,
            'urlAlias-notFound',
        ];

        if ($urlAlias instanceof UrlAlias) {
            $cacheTags[] = 'urlAlias-' . $urlAlias->id;
        }
        
        $this->cache->invalidateTags($cacheTags);
```

But going a bit further, `createCustomUrlAlias` should always return UrlAlias, so here we should adapt tests I thin (if possible).

Reply:
Resolved in https://github.com/ezsystems/ezpublish-kernel/pull/3062/commits/dd8975e3e8af75987d3c0552f5ef1b5dd21a2b1b